python/semantic_kernel/connectors/memory/azure_ai_search.py

```diff
@@ -0,0 +1,724 @@
+                    name=field.storage_property_name or field.name,
+                    type=type_,
+                    filterable=field.is_indexed or field.is_full_text_indexed,
+                    # searchable is set first on the value of is_full_text_searchable,
+                    # if it is None it checks the field type, if text then it is searchable
+                    searchable=type_ in ("Edm.String", "Collection(Edm.String)")
+                    if field.is_full_text_indexed is None
+                    else field.is_full_text_indexed,
+                    sortable=True,
+                    hidden=False,
+                )
+            )
+        elif isinstance(field, VectorStoreRecordKeyField):
+            fields.append(
+                SimpleField(
+                    name=field.storage_property_name or field.name,
+                    type="Edm.String",  # hardcoded, only allowed type for key
+                    key=True,
+                    filterable=True,
+                    searchable=True,
+                )
+            )
+        elif isinstance(field, VectorStoreRecordVectorField):
+            if not field.property_type:
+                logger.debug(f"Field {field.name} has not specified type, defaulting to Collection(Edm.Single).")
+            if field.index_kind not in INDEX_ALGORITHM_MAP:
+                raise VectorStoreOperationException(f"{field.index_kind} not supported in Azure AI Search.")
+            if field.distance_function not in DISTANCE_FUNCTION_MAP:
+                raise VectorStoreOperationException(f"{field.distance_function} not supported in Azure AI Search.")
+
+            profile_name = f"{field.storage_property_name or field.name}_profile"
+            algo_name = f"{field.storage_property_name or field.name}_algorithm"
+            fields.append(
+                SearchField(
+                    name=field.storage_property_name or field.name,
+                    type=TYPE_MAP_VECTOR[field.property_type or "default"],
+                    searchable=True,
+                    vector_search_dimensions=field.dimensions,
+                    vector_search_profile_name=profile_name,
+                    hidden=False,
+                )
+            )
+            search_profiles.append(
+                VectorSearchProfile(
+                    name=profile_name,
+                    algorithm_configuration_name=algo_name,
+                )
+            )
+            algo_class, algo_params = INDEX_ALGORITHM_MAP[field.index_kind]
+            distance_metric = DISTANCE_FUNCTION_MAP[field.distance_function]
+            search_algos.append(algo_class(name=algo_name, parameters=algo_params(metric=distance_metric)))
+    return SearchIndex(
+        name=collection_name,
+        fields=fields,
+        vector_search=AZSVectorSearch(profiles=search_profiles, algorithms=search_algos),
+        encryption_key=encryption_key,
+    )
+
+
+@experimental
+class AzureAISearchCollection(
```

Comment:
Can you help me understand this new "separation of concerns" where we have a "memory" directory and a "memory_stores" directory? The interactions with the memory indexes are housed in `memory` and the backing store logic is in `memory_stores`? Can't everything live under the respective folder inside of `memory`? You've been a proponent of reducing the number of files in SK. As an example, seems like all Redis related content could live in `semantic_kernel/connectors/redis.py` and then all Redis-related imports look like `from semantic_kernel.memory import RedisMemoryStore, RedisCollection, ...`